src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -30,7 +30,7 @@ internal class SubstitutedMethodSymbol : WrappedMethodSymbol
         private ImmutableArray<MethodSymbol> _lazyExplicitInterfaceImplementations;
         private OverriddenOrHiddenMembersResult _lazyOverriddenOrHiddenMembers;
 
-        private int _hashCode; // computed on demand
+        private int? _hashCode; // computed on demand
```

Comment:
>int? [](start = 16, length = 4)

This probably makes the code that initializes this field not thread-safe. The assignment is no longer guaranteed to be atomic. I am not sure if the change is warranted, it is unlikely that definition's hash code is 0. We can even assert that in the ComputeHashCode.